lib/discourse_chat_integration/chat_integration_reference_post.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module DiscourseChatIntegration
+  # This _emulates_ a post, but is not a real post
+  class ChatIntegrationReferencePost
+    def initialize(context)
+      @context = context
+      @user = context["user"]
+      @topic = context["topic"]
+      @kind = context["kind"]
+      @raw = context["raw"] if context["raw"].present?
+      @full_url = (@topic.posts.empty? ? @topic.full_url : @topic.posts.first.full_url)
```

Comment:
I think maybe move this logic down into the method rather than having in intiializer